libs/csm/rest/csm_rest_capacity.py

```diff
@@ -47,32 +46,104 @@ def __init__(self):
         self.cryptogen = SystemRandom()
 
     @RestTestLib.authenticate_and_login
-    def get_capacity_usage(self):
+    def get_capacity_usage(self, endpoint_param=None ,auth_header=None):
         """Get the system capacity usage
 
         :return [obj]: Json response
         """
-        try:
-            # Building request url
-            self.log.info("Reading System Capacity...")
-            endpoint = self.config["capacity_endpoint"]
-            self.log.info("Endpoint for reading capacity is %s", endpoint)
+        if auth_header is None:
+            headers = self.headers
+        else:
+            headers = auth_header
+
+        self.log.info("Reading System Capacity...")
+        endpoint = self.config["capacity_endpoint"]
+        if endpoint_param is not None:
+            endpoint = self.config["capacity_endpoint"] + "/" + endpoint_param
+        self.log.info("Endpoint for reading capacity is %s", endpoint)
+        # Fetching api response
+        response = self.restapi.rest_call(request_type="get", endpoint=endpoint,
+                                          headers=headers)
+        return response
+
+    def get_degraded_capacity(self, auth_header=None):
+        """
+        Get degraded capacity from CSM
+        :param endpoint_param: which endpoint to check for parameters
+        :return : Rest output response
+        """
+        response = self.get_capacity_usage(endpoint_param="bytecount", auth_header=auth_header)
+        return response
+
+    def verify_get_degraded_capacity(self, auth_header=None, expected_response=HTTPStatus.OK):
```

Comment:
can we not combine these get and verify?

Reply:
We always have one basic function and all verification is build on top of it within CSM.